src/crypto/index.js

```diff
@@ -271,6 +283,28 @@ function _maybeUploadOneTimeKeys(crypto) {
 
     crypto._lastOneTimeKeyCheck = now;
 
+    // We need to keep a pool of one time public keys on the server so that
+    // other devices can start conversations with us. But we can only store
+    // a finite number of private keys in the olm Account object.
+    // To complicate things further then can be a delay between a device
+    // claiming a public one time key from the server and it sending us a
+    // message. We need to keep the corresponding private key locally until
+    // we receive the message.
+    // But that message might never arrive leaving us stuck with duff
+    // private keys clogging up our local storage.
+    // So we need some kind of enginering compromise to balance all of
+    // these factors.
+
+    // We then check how many keys we can store in the Account object.
```

Comment:
this comment doesn't make any sense any more, because you have removed the previous sentence ("We first find out...").

I suggest you just `s/We then //`